### ISSUE TITLE
Release build test failures:

### ISSUE THREAD

**Issue by michael-loki** [loki-project/loki-network#567]
**Describe the bug**
In release builds on linux, two tests fail:

```
[----------] 2 tests from LinkLayerTest
[ RUN      ] LinkLayerTest.TestUTPAliceRenegWithBob
../test/link/test_llarp_link.cpp:256: Failure
Value of: Bob.gotLIM
  Actual: false
Expected: true
[  FAILED  ] LinkLayerTest.TestUTPAliceRenegWithBob (5011 ms)
[ RUN      ] LinkLayerTest.TestUTPAliceConnectToBob
../test/link/test_llarp_link.cpp:345: Failure
Value of: Bob.gotLIM
  Actual: false
Expected: true
[  FAILED  ] LinkLayerTest.TestUTPAliceConnectToBob (5019 ms)
[----------] 2 tests from LinkLayerTest (10030 ms total)
```

**To Reproduce**
Build in release mode 

**Screenshots Or Logs**
see above

**Device and Operating system (please complete the following information):**
 - OS: alpine linux
-  Device: inside docker container
 - Lokinet Version number or Git commit hash: 81bf00f9bc7ec88c175fc083f4f53dd74f9e2591

**Comment by despair86** [comment]
see #400
fix: make sure NDEBUG is not defined in `CMAKE_*_FLAGS_RELEASE`, better yet, make sure those are blank, we already use the optimiser in non-debug builds: `-O3 -march...`, in fact, the Windows build is delivered with the optimiser turned ON, but with the cmake release flags cleared to work around this bug for now

**Comment by michael-loki** [comment]
> we already use the optimiser in non-debug builds

every build is currently debug though, which unsets them: https://github.com/loki-project/loki-network/blob/81bf00f9bc7ec88c175fc083f4f53dd74f9e2591/CMakeLists.txt#L230-L236

**Comment by majestrate** [comment]
https://github.com/loki-project/loki-network/blob/353021b671bfe830b35165c6e80fa34da8e3497d/llarp/utp/session.cpp#L696

well there the problem

**Comment by michael-loki** [comment]
fixed in #568
